[18 India] Add Commodity Destination Table Below Map

EdPovi opened the pull request:
Fixes #11066, #11068, #11128 

## Before clicking "Create"

- [x] Branch is derived from the latest `master`
- [x] Add the `pins` or `archive_alpha_games` label if this change will break existing games
- [x] Code passes linter with `docker compose exec rack rubocop -a`
- [x] Tests pass cleanly with `docker compose exec rack rake`

## Implementation Notes
NOTE: This PR should not break existing game. 

### Explanation of Change
Added table for Commodity Destination Bonus and Connection Bonus using table like in RoyalGorge.
Also modified the look of Large Icons as the circle "decoration" looks a bit like a city.  

### Screenshots
![image](https://github.com/user-attachments/assets/a9e2cba9-f384-4a19-b5ce-8720a4601434)
![image](https://github.com/user-attachments/assets/3e148cc2-f87f-4cf8-8895-3a22050fb367)
![image](https://github.com/user-attachments/assets/12d0f4a8-38a1-4597-94b8-fcf389c41350)

### Any Assumptions / Hacks
Added commodity price icons, but the hexes are too crowded to add 2-3 icons to the city hexes.  Also the `LargeIcons` feature doesn't work well with more than one Icon.  Plan to customize or create new class to display larger icons.

ollybh commented on lib/engine/game/g_18_india/game.rb:
```diff
@@ -1230,6 +1230,222 @@ def price_movement_chart
+              { text: 'Connection Bonus', props: { attrs: { colspan: 10 }, style: { **cell_style, backgroundColor: green } } },
+            ],
+            [
+              { text: 'Delhi (G8) <=> Kocchi (G36)', props: { style: cell_style } },
+              { text: format_currency(100), props: { style: cell_style } },
+            ],
+            [
+              { text: 'Karachi (A16) <=> Chennai (K30)', props: { style: cell_style } },
+              { text: format_currency(80), props: { style: cell_style } },
+            ],
+            [
+              { text: 'Lahore (D3) <=> Kolkata (P17)', props: { style: cell_style } },
+              { text: format_currency(80), props: { style: cell_style } },
+            ],
+            [
+              { text: 'Nepal (M10) <=> Mumbai (D23)', props: { style: cell_style } },
```
Unicode support should be universal these days, so we don't have to use Ascii compositions for arrows:
```suggestion
              { text: 'Delhi (G8) ⟷ Kocchi (G36)', props: { style: cell_style } },
              { text: format_currency(100), props: { style: cell_style } },
            ],
            [
              { text: 'Karachi (A16) ⟷ Chennai (K30)', props: { style: cell_style } },
              { text: format_currency(80), props: { style: cell_style } },
            ],
            [
              { text: 'Lahore (D3) ⟷ Kolkata (P17)', props: { style: cell_style } },
              { text: format_currency(80), props: { style: cell_style } },
            ],
            [
              { text: 'Nepal (M10) ⟷ Mumbai (D23)', props: { style: cell_style } },
```

ollybh commented:
This looks good, but I've got a couple of suggestions:

1. The commodity icons would look better with no surrounding shape. Return a `{ shape: :none }` from `decorate_markers` and update the case statement in `View::Game::Part::LargeIcons.render_part` to handle this.
2. Add an extra column to the commodities table to show the hex coordinates of each city.

EdPovi commented:
> 1. The commodity icons would look better with no surrounding shape. Return a `{ shape: :none }` from `decorate_markers` and update the case statement in `View::Game::Part::LargeIcons.render_part` to handle this.

I actually did almost exactly that at first, and then reverted to minimize Core changes.
This gives me the support to make this change and maybe even extend it further to support multiple large icons on a single hex.  This requires looking at setting alternate locations based on number of large icons.  But I think I can use Icon information and the decorate info to control what happens. 

> 2. Add an extra column to the commodities table to show the hex coordinates of each city.

This should be easy enough to do.

EdPovi commented:
Updated Table and added hex icons using by adapting `large_icons` to support up to 3 large icons in an empty hex. 

![image](https://github.com/user-attachments/assets/34e552de-9a87-494f-930b-8c2188e59955)
![image](https://github.com/user-attachments/assets/99a1f187-4a71-42e5-9fb5-419d52c55f29)

ollybh commented:
This has improved the appearance of the icons a lot.

The one thing I'm not sure is right is the `loc:hex` attribute on the large icon tile part. If someone created a tile which had some large icons with this set and some without, then only the ones with `loc:hex` would be rendered and the others would be not shown. This could cause confusion.

Does this render at all sensibly if you just add the large icons without changing the code for their positions? It might be worth getting the rest of this pull request merged without the `render_hex_icons` bit, and separating that into a separate pull request.

EdPovi commented:
> Does this render at all sensibly if you just add the large icons without changing the code for their positions? It might be worth getting the rest of this pull request merged without the `render_hex_icons` bit, and separating that into a separate pull request.

Unfortunately, it doesn't the original code for `large_icons` doesn't support more than one icon, and they noted that limitation in the code: `# multiple large icons not supported`

Also, using `loc:hex` is intended to be used only for icons that will be rendered that way.  Using both types would not work, which is why set the code path to use one or the other render path.  Also, this is intended only for **empty** hexes as the position make no consideration for any other items on that hex.

ollybh commented:
> Unfortunately, the original code for `large_icons` doesn't support more than one icon, and they noted that limitation in the code: `# multiple large icons not supported`. Therefore this alternate path is necessary if I place more than one on hex.

The code does handle multiple large icons (there's an iterator in `render_part`, it just doesn't handle them well:
![image](https://github.com/user-attachments/assets/07268516-9105-4f4b-8a32-bce2aff8716f)

ollybh commented:
Could these icons be added with this sub-optimal rendering? I'm happy to approve the rest of this pull request, but would need one of the more experienced developers to approve the changes to the large icon positioning, so it might be better split into its own pull request.

EdPovi commented:
> Could these icons be added with this sub-optimal rendering? I'm happy to approve the rest of this pull request, but would need one of the more experienced developers to approve the changes to the large icon positioning, so it might be better split into its own pull request.

I understand and I can split the alternate rendering to a separate PR.
